nexus/db-queries/src/db/queries/network_interface.rs

```diff
@@ -292,80 +289,76 @@ fn decode_database_error(
         // limit total number of interfaces per resource to a maximum number.
-        ConnectionError::Query(Error::DatabaseError(
+        DieselError::DatabaseError(
             DatabaseErrorKind::CheckViolation,
             ref info,
-        )) if info.message() == NO_SLOTS_AVAILABLE_ERROR_MESSAGE => {
+        ) if info.message() == NO_SLOTS_AVAILABLE_ERROR_MESSAGE => {
             InsertError::NoSlotsAvailable
         }
 
         // If the MAC allocation subquery fails, we'll attempt to insert NULL
         // for the `mac` column. This checks that the non-NULL constraint on
         // that column has been violated.
-        ConnectionError::Query(Error::DatabaseError(
+        DieselError::DatabaseError(
             DatabaseErrorKind::NotNullViolation,
             ref info,
-        )) if info.message() == MAC_EXHAUSTION_ERROR_MESSAGE => {
+        ) if info.message() == MAC_EXHAUSTION_ERROR_MESSAGE => {
             InsertError::NoMacAddrressesAvailable
         }
 
         // This catches the error intentionally introduced by the
         // `push_ensure_unique_vpc_subnet_expression` subquery, which generates
         // a UUID parsing error if the resource has another interface in the VPC
         // Subnet of the one we're trying to insert.
-        ConnectionError::Query(Error::DatabaseError(
-            DatabaseErrorKind::Unknown,
-            ref info,
-        )) if info.message() == NON_UNIQUE_VPC_SUBNET_ERROR_MESSAGE => {
+        DieselError::DatabaseError(DatabaseErrorKind::Unknown, ref info)
+            if info.message() == NON_UNIQUE_VPC_SUBNET_ERROR_MESSAGE =>
+        {
             InsertError::NonUniqueVpcSubnets
         }
 
         // This catches the UUID-cast failure intentionally introduced by
         // `push_instance_state_verification_subquery`, which verifies that
         // the instance is actually stopped when running this query.
-        ConnectionError::Query(Error::DatabaseError(
-            DatabaseErrorKind::Unknown,
-            ref info,
-        )) if info.message() == INSTANCE_BAD_STATE_ERROR_MESSAGE => {
+        DieselError::DatabaseError(DatabaseErrorKind::Unknown, ref info)
+            if info.message() == INSTANCE_BAD_STATE_ERROR_MESSAGE =>
+        {
             assert_eq!(interface.kind, NetworkInterfaceKind::Instance);
             InsertError::InstanceMustBeStopped(interface.parent_id)
         }
         // This catches the UUID-cast failure intentionally introduced by
         // `push_instance_state_verification_subquery`, which verifies that
         // the instance doesn't even exist when running this query.
-        ConnectionError::Query(Error::DatabaseError(
-            DatabaseErrorKind::Unknown,
-            ref info,
-        )) if info.message() == NO_INSTANCE_ERROR_MESSAGE => {
+        DieselError::DatabaseError(DatabaseErrorKind::Unknown, ref info)
+            if info.message() == NO_INSTANCE_ERROR_MESSAGE =>
```

Comment:
Don't need `ref` here these days (here or in the rest of the file)

Reply:
I do need *one* of them in a case where we re-use the `err`  we're matching on, but I removed the rest